integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -372,8 +391,10 @@ def test_cast_neg_to_decimal_err():
     data_gen = _decimal_gen_7_7
     if is_before_spark_322():
         exception_content = "Decimal(compact,-120000000,20,0}) cannot be represented as Decimal(7, 7)"
-    elif is_databricks113_or_later() or not is_before_spark_340():
+    elif is_databricks113_or_later() or not is_before_spark_340() and is_before_spark_400():
         exception_content = "[NUMERIC_VALUE_OUT_OF_RANGE] -12 cannot be represented as Decimal(7, 7)"
+    elif not is_before_spark_400():
+        exception_content = "[NUMERIC_VALUE_OUT_OF_RANGE.WITH_SUGGESTION]  -12 cannot be represented as Decimal(7, 7)"
```

Comment:
Silly question:  Is that double space before `-12` really part of the exception?
```suggestion
        exception_content = "[NUMERIC_VALUE_OUT_OF_RANGE.WITH_SUGGESTION] -12 cannot be represented as Decimal(7, 7)"
```

Reply:
Keen eye! 
Yes, as absurd as it seems, it is a part of the actual exception being thrown

Reply:
I can't wait for this to be "corrected" on Apache Spark. Is there a utility that we should be using for this error message?